tests/common/osutil/test_default.py

```diff
@@ -737,59 +738,97 @@ def test_enable_firewall_should_set_up_the_firewall(self):
+                                 "The sixth command should delete the owner accept rule: {0}".format(
+                                     mock_iptables.command_calls[5]))
+                self.assertEqual(mock_iptables.command_calls[6], delete_conntrack_drop_command,
+                                 "The seventh command should delete the conntrack accept rule : {0}".format(
+                                     mock_iptables.command_calls[6]))
+                self.assertEqual(mock_iptables.command_calls[7], accept_tcp_command,
+                                "The eighth command should add the dns tcp accept rule")
+                self.assertEqual(mock_iptables.command_calls[8], accept_command, "The ninth command should add the accept rule")
+                self.assertEqual(mock_iptables.command_calls[9], drop_add_command, "The tenth command should add the drop rule")
 
                 self.assertTrue(osutil._enable_firewall, "The firewall should not have been disabled")
 
     def test_enable_firewall_should_not_use_wait_when_iptables_does_not_support_it(self):
         with TestOSUtil._mock_iptables(version=osutil._IPTABLES_LOCKING_VERSION - 1) as mock_iptables:
             with patch.object(osutil, '_enable_firewall', True):
                 # fail the rule check to force enable of the firewall
-                mock_iptables.set_command(
-                    osutil.get_firewall_drop_command(mock_iptables.wait, AddFirewallRules.CHECK_COMMAND,
-                                                     mock_iptables.destination), exit_code=1)
+                mock_iptables.set_command(AddFirewallRules.get_accept_tcp_rule(AddFirewallRules.CHECK_COMMAND, mock_iptables.destination,
+                                                                               wait=mock_iptables.wait), exit_code=1)
+                mock_iptables.set_command(AddFirewallRules.get_accept_command(AddFirewallRules.CHECK_COMMAND, mock_iptables.destination, mock_iptables.uid,
+                                                                              wait=mock_iptables.wait), exit_code=1)
+                mock_iptables.set_command(AddFirewallRules.get_drop_command(AddFirewallRules.CHECK_COMMAND, mock_iptables.destination,
+                                                                            wait=mock_iptables.wait), exit_code=1)
 
                 success = osutil.DefaultOSUtil().enable_firewall(dst_ip=mock_iptables.destination, uid=mock_iptables.uid)
 
                 self.assertTrue(success, "Enabling the firewall was not successful")
-                # Exactly 4 calls have to be made.
-                # First is the DROP rule check which was mocked to fail, Second, third and forth
-                # to Append the three IPTable rules
-                self.assertEqual(len(mock_iptables.command_calls), 4,
+                # Exactly 8 calls have to be made.
+                # First check rule, delete 4 rules.
```

Comment:
How come delete 4 rules? Shouldn't it be 3? 

Additionally, NIT on the full stop - 

```suggestion
                # First check rule, delete 4 rules,
```

Reply:
we had legacy rule that was used <=2.2.25